### PR TITLE
Fixes #23151: Allow booleans for condition fields

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/23151

`true` and `false` are special value, let's accept them as boolean too.

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: policies/rudderc/tests/cases/general/ntp/technique.ps1 [hunk 1]
```diff
@@ -35,7 +35,7 @@
         TechniqueName = $techniqueName
     }
     
-    $class = ""debian""
+    $class = ""false""
```
Comment (amousset) [inline policies/rudderc/tests/cases/general/ntp/technique.ps1:38]: these double quotes look dubious to me

Comment (Fdall) [reply policies/rudderc/tests/cases/general/ntp/technique.ps1:38]: Double double quotes does not exist in powershell

**Review by Fdall — APPROVED** [review]

**Pull request merged**